### PR TITLE
Add task

### PR DESCRIPTION
**Pull request by LucieLDa**
Add the addTaskComponent
This component is a child of the component taskList.
Allows the user to add a task to the task list. Has two fields for date and description.
Add the modalComponent
This component can be used for any modal the this projet needs. Can be opened from a parent component and closes when clicking on the background. Iy's contnent can be determined in it's parent component.
Minor ajustements in TaskList and style.scss to accomodate the two new components.

### CODE REVIEW

**Review by MoubarkiS — APPROVED** [review]
Clean Code

File: src/app/components/add-task/add-task.component.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+@Component({
+  selector: 'app-add-task',
+  standalone: true,
+  imports: [ModalComponent, FormsModule, CommonModule],
+  templateUrl: './add-task.component.html',
+  styleUrl: './add-task.component.scss'
+})
+export class AddTaskComponent {
+  taskModalOpen : boolean = false;
```
Comment (Benja-san) [inline src/app/components/add-task/add-task.component.ts]: consider rename this variable witrh "is" prefix

File: src/app/components/add-task/add-task.component.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  }
+
+  openAddTaskModal() : void {
+    this.taskModalOpen = true;
+  }
+
+  //this function is to make the textarea expend when the user types out the task's description
+  autogrow(){
+    let textArea = document.getElementById("task-name")       
+    //textArea!.style.overflow = 'hidden';
```
Comment (Benja-san) [inline src/app/components/add-task/add-task.component.ts]: death code

File: src/app/components/task-list/task-list.component.html [hunk 3]
```diff
@@ -14,5 +14,8 @@ <h1>Liste des tâches:</h1>
         <li *ngFor="let task of taskList" >
             <app-task [task]="task"></app-task>
         </li>
+        <li>
```
Comment (Benja-san) [inline src/app/components/task-list/task-list.component.html]: put bellow ul

**Review by Benja-san — CHANGES_REQUESTED** [review]

**Review by Benja-san — APPROVED** [review]

**Pull request merged**